base/collection.go

```diff
@@ -985,3 +986,71 @@ func (t *SGRawTranscoder) Encode(value interface{}) ([]byte, uint32, error) {
 	return gocb.NewRawBinaryTranscoder().Encode(value)
 
 }
+
+// GetGoCBAgent returns the underlying agent from gocbcore
+func (c *Collection) getGoCBAgent() (*gocbcore.Agent, error) {
+	return c.Bucket().Internal().IORouter()
+
+}
+
+func (c *Collection) getBucketOpDeadline() time.Time {
+	opTimeout := DefaultGocbV2OperationTimeout
+	configOpTimeout := c.Spec.BucketOpTimeout
+	if configOpTimeout != nil {
+		opTimeout = *configOpTimeout
+	}
+	return time.Now().Add(opTimeout)
+}
+
+// getCollectionID returns the gocbcore CollectionID for the current collection
+func (c *Collection) getCollectionID() (uint32, error) {
+	// return cached value if present
+	if c.collectionID != nil {
```

Comment:
Do we have any scenarios where we'd have concurrent invocation of getCollectionID that would merit using atomics to read/write c.collectionID?

Reply:
We definitely could so I made this an Atomic.Value